internal/stateindex/index_entries.go

```diff
@@ -13,12 +13,20 @@ import (
 	"github.com/IBM-Blockchain/bcdb-server/pkg/types"
 )
 
-// IndexDBPrefix is the prefix added to each user database to create an index
-// database for that user database
 const (
-	IndexDBPrefix  = "_index_"
-	PositiveNumber = "p"
-	NegativeNumber = "n"
+	// indexDBPrefix is the prefix added to each user database to create an index
+	// database for that user database
+	indexDBPrefix = "_index_"
+	// positiveNumber and negativeNumber are metadata that denote whether the value being
+	// index is positive or negative
+	positiveNumber = "p"
+	negativeNumber = "n"
+)
+
+const (
+	Beginning = iota
+	Existing
+	Ending
```

Comment:
Could you say a few words on how this is used and why is it needed? 
It is difficult to understand at this stage.... This will probably become clearer in future commits... nevertheless, it would also be easier for the reader if there would be a comment that describes the index entry structure...
Why the iota? I only see `Existing` used in the code so far. Is there an intention to add more values, or to use `Beginning`  and `Ending`?